File: plugins/methods/src/com/intellij/methods/editor/MethodsEditorProvider.java
package com.intellij.methods.editor;

import com.intellij.methods.fs.MethodsVirtualFileSystem;
import com.intellij.openapi.fileEditor.FileEditor;
import com.intellij.openapi.fileEditor.FileEditorPolicy;
import com.intellij.openapi.fileEditor.FileEditorProvider;
import com.intellij.openapi.fileEditor.TextEditorProvider;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

public final class MethodsEditorProvider implements FileEditorProvider {
  @Override
  public boolean accept(@NotNull Project project, @NotNull VirtualFile file) {
    return file.getFileSystem() instanceof MethodsVirtualFileSystem;
  }

  @Override
  public @NotNull FileEditor createEditor(@NotNull Project project, @NotNull VirtualFile file) {
    return TextEditorProvider.getInstance().createEditor(project, file);
  }

  @Override
  public @NotNull String getEditorTypeId() {
    return "methods-view";
  }

  @Override
  public @NotNull FileEditorPolicy getPolicy() {
    return FileEditorPolicy.HIDE_DEFAULT_EDITOR;
  }
}

File: plugins/methods/src/com/intellij/methods/editor/MethodsEditorProvider.kt
package com.intellij.methods.editor

import com.intellij.methods.fs.MethodsVirtualFileSystem
import com.intellij.openapi.fileEditor.FileEditor
import com.intellij.openapi.fileEditor.FileEditorPolicy
import com.intellij.openapi.fileEditor.FileEditorProvider
import com.intellij.openapi.fileEditor.TextEditorProvider
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile

/**
 * Fallback editor that delegates to the standard text editor.
 */
class MethodsEditorProvider : FileEditorProvider {
    override fun accept(project: Project, file: VirtualFile): Boolean =
        file.fileSystem is MethodsVirtualFileSystem

    override fun createEditor(project: Project, file: VirtualFile): FileEditor =
        TextEditorProvider.getInstance().createEditor(project, file)

    override fun getEditorTypeId(): String = "methods-view"

    override fun getPolicy(): FileEditorPolicy = FileEditorPolicy.HIDE_DEFAULT_EDITOR
}

